client/signers.go

```diff
@@ -89,6 +97,14 @@ func (s *BasicSigner) Sign(req *http.Request) error {
 	return nil
 }
 
+// SignWebSocket adds the basic auth header to the WebSocket dial config
+func (s *WebSocketSigner) SignWebSocket(cfg *websocket.Config) error {
```

Comment:
That does not work, you need to have one struct per security scheme each implementing the `WebSocketSigner` interface similarly to how it's done with the existing `Signer` interface.